lib/irclogger/viewer_helpers.rb

```diff
@@ -3,13 +3,44 @@
 module IrcLogger
   module ViewerHelpers
     include Rack::Utils
+    
+    CHANNEL_ESCAPE = {
+      '~' => '~~',
+      '#' => '~h~',
+      '/' => '~s~',
+      '%' => '~p~',
+      '\\' => '~r~',
+      '?' => '~q~',
+      '`' => '~a~',
+      '<' => '~l~',
+      '>' => '~g~',
+      '|' => '~b~',
+      '{' => '~o~',
+      '}' => '~c~',
+      '^' => '~x~',
+      '"' => '~d~'
+    }
+    
+    def escape_url(url)
+      url.gsub(/(.)/) { |m| CHANNEL_ESCAPE.key?(m) ? CHANNEL_ESCAPE[m] : m }
+    end
+    
+    def unescape_url(url)
+      inv = CHANNEL_ESCAPE.invert 
```

Comment:
This cound be lifted to a constant, to avoid wasting time inverting the hash on every call.